public/app/features/plugins/plugin_loader.ts

```diff
@@ -94,7 +96,17 @@ export async function importPluginModule({
     }
   }
 
-  let modulePath = resolveModulePath(path);
+  const modulePath = resolveModulePath(path);
+  const resolvedModule = System.resolve(modulePath);
+  const integrityMap = System.getImportMap().integrity;
+
+  if (moduleHash && integrityMap && !integrityMap[resolvedModule]) {
```

Comment:
since all this functionality is behind a feature flag, we need to also make sure the frontend code only executes is the feature flag is on to prevent any possible side effects

Reply:
Sounds like a good suggestion by @academo 👍🏻

Reply:
Thanks for the suggestion. I've updated the logic in this commit: https://github.com/grafana/grafana/pull/93024/commits/4c9a1cf6ac387437ff6673d4da2ec86a13c79c97